trunk/SUAVE/Methods/Weights/Buildups/Common/wing.py

```diff
@@ -0,0 +1,286 @@
+    box = box * chord[0]               # Scale by Chord Length
+
+        # Use Shoelace Formula to calculate box area
+
+    torsionArea = 0.5*np.abs(np.dot(box[:,0],np.roll(box[:,1],1))-
+        np.dot(box[:,1],np.roll(box[:,0],1)))
+
+    torsionLength = np.sum(np.sqrt(np.sum(np.diff(box,axis=0)**2,axis=1)))
+
+    # Bending
+
+    box = coord                     # Box Initally Matches Airfoil
+    box = box[box[:,0]<=fwdWeb[1]]  # Inlcude Only Parts Fwd of Aft Fwd Spar
+    box = box[box[:,0]>=fwdWeb[0]]  # Include Only Parts Aft of Fwdmost Spar
+    seg.append(box[box[:,1]>np.mean(box[:,1])]*chord[0])   # Upper Fwd Segment
+    seg.append(box[box[:,1]<np.mean(box[:,1])]*chord[0])   # Lower Fwd Segment
+
+    # Drag
+
+    box = coord                     # Box Initally Matches Airfoil
+    box = box[box[:,0]<=aftWeb[1]]  # Inlcude Only Parts Fwd of Aftmost Spar
+    box = box[box[:,0]>=aftWeb[0]]  # Include Only Parts Aft of Fwd Aft Spar
+    seg.append(box[box[:,1]>np.mean(box[:,1])]*chord[0])   # Upper Aft Segment
+    seg.append(box[box[:,1]<np.mean(box[:,1])]*chord[0])   # Lower Aft Segment
+
+    # Bending/Drag Inertia
+
+    flapInertia = 0
+    flapLength  = 0
+    dragInertia = 0
+    dragLength  = 0
+
+    for i in range(0,4):
+        l = np.sqrt(np.sum(np.diff(seg[i],axis=0)**2,axis=1))    # Segment lengths
+        c = (seg[i][1::]+seg[i][0:-1])/2                         # Segment centroids
+
+        if i<2:
+            flapInertia += np.abs(np.sum(l*c[:,1]**2))   # Bending Inertia per Unit Thickness
+            flapLength  += np.sum(l)
+        else:
+            dragInertia += np.abs(np.sum(l*c[:,0]**2))   # Drag Inertia per Unit Thickness
+            dragLength  += np.sum(l)
+
+
+    # Shear
+
+    box = coord                     # Box Initially Matches Airfoil
+    box = box[box[:,0]<=fwdWeb[1]]  # Include Only Parts Fwd of Aft Fwd Spar
+    z = np.zeros(2)
+    z[0] = np.interp(fwdWeb[0],box[box[:,1]>0,0],box[box[:,1]>0,1])*chord[0]  # Upper Surface of Box at Fwdmost Spar
+    z[1] = np.interp(fwdWeb[0],box[box[:,1]<0,0],box[box[:,1]<0,1])*chord[0]  # Lower Surface of Box at Fwdmost Spar
+    h = np.abs(z[0] - z[1])                 # Height of Box at Fwdmost Spar
+
+    # Skin
+
+    box = coord * chord             # Box Initially is Airfoil Scaled by Chord
+    skinLength = np.sum(np.sqrt(np.sum(np.diff(box,axis=0)**2,axis=1)))
+    A = 0.5*np.abs(np.dot(box[:,0],np.roll(box[:,1],1))-
+        np.dot(box[:,1],np.roll(box[:,0],1)))   # Box Area via Shoelace Formula
+
+    #---------------------------------------------------------------------------
+    # Structural Calculations
+    #---------------------------------------------------------------------------
+
+    # Calculate Skin Weight Based on Torsion
+
+    tTorsion = My*dx/(2*bidirectional_carbon_fiber().ultimate_shear_strength*torsionArea)                                    # Torsion Skin Thickness
```

Comment:
unpack things

Reply:
Changed to unpack of material properties.